### PR TITLE
[VUX-976] Missing title fix

### PR DESCRIPTION
**Pull request by richardbata**
## 🖼 Context
The layout property is not yet fully supported on mobile platforms, leading top level title properties not within the layout to be displayed as page headings. This issue will be solved correctly, but until then I'm indroducing a fix on web to make is consistent with the behaviour on mobile.

Essentially the JSON below result in double titles on mobile, but only a single one on web. Web should be correct, as specifying a `layout` property should override all default layouts. Until the issue is solved on mobile, web is introduction this fix.
image

## ✅ Checklist

- [ ] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [ ] Changes are tested and all tests pass
- [ ] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/flow/DynamicFlow.js [hunk 1]
```diff
@@ -40,7 +40,16 @@ const getComponents = (step) => {
     return [];
   }
 
-  const layout = step.layout ? step.layout : convertStepToLayout(step);
+  let layout = step.layout ? step.layout : convertStepToLayout(step);
+
+  // This condition is meant as a temporary hack until mobile platforms will be refactored to fully support the layout property
+  if (step.layout && step.title) {
+    const topStep = {
```
Comment (StevePole) [inline packages/dynamic-flows/src/flow/DynamicFlow.js:47]: We should do this within the convertStepToLayout function, that's effectively designed for backwards compatibility with steps that don't have layouts.

File: packages/dynamic-flows/src/flow/DynamicFlow.js [hunk 2]
```diff
@@ -40,7 +40,16 @@ const getComponents = (step) => {
...
+      type: step.type,
```
Comment (StevePole) [inline packages/dynamic-flows/src/flow/DynamicFlow.js:48]: Bit confused by this, the step type is not going to be a valid layout component type.

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.js [hunk 3]
```diff
@@ -3,6 +3,10 @@ function convertStepToLayout(step) {
     throw new Error('Missing step type');
   }
 
+  if (step.layout) {
```
Comment (StevePole) [inline packages/dynamic-flows/src/flow/layoutService/LayoutService.js:6]: Could you just add back the note about why this exists?

Comment (richardbata) [reply packages/dynamic-flows/src/flow/layoutService/LayoutService.js:7]: Done

**Review by StevePole — APPROVED** [review]

**Pull request merged**